flake.nix

```diff
@@ -190,6 +193,47 @@
                 if final.system == "aarch64-darwin"
                 then overrideCabal hsuper.ormolu (_: { enableSeparateBinOutput = false; })
                 else hsuper.ormolu;
+
+              # Fourmolu needs a handful of patches to build on GHC 9.4.
+              fourmoluHEAD =
+                appendPatches
+                (doJailbreak (if final.system == "aarch64-darwin"
+                              then overrideCabal hsuper.fourmolu_0_8_2_0 (_: { enableSeparateBinOutput = false; })
+                              else hsuper.fourmolu_0_8_2_0))
+                [
+                  # The GHC 9.4 pull request builds upon these unpublished changes,
+                  # which is why we include them.
+                  #
+                  # Also, there is no PR for these changes.  The fourmolu maintainers
```

Comment:
this isn't true, for example the first commit is this pr: https://github.com/fourmolu/fourmolu/pull/229

Reply:
OK, but this is just for testing. We'll have to release a new Fourmolu before this PR can build outside of Nix.